src/main/java/com/linecorp/armeria/server/logging/DropwizardMetricCollectingService.java

```diff
@@ -59,13 +86,41 @@
     Function<Service<? super I, ? extends O>, DropwizardMetricCollectingService<I, O>> newDecorator(
             MetricRegistry metricRegistry, String metricNamePrefix) {
 
-        return service -> new DropwizardMetricCollectingService<>(
-                service, metricRegistry, metricNamePrefix);
+        requireNonNull(metricNamePrefix, "metricNamePrefix");
+        return newDecorator(metricRegistry, (ctx, req) -> defaultMetricName(ctx, req, metricNamePrefix));
+    }
+
+    private static String defaultMetricName(
+            ServiceRequestContext ctx, RequestLog req, String metricNamePrefix) {
+
+        final String hostname = ctx.virtualHost().defaultHostname().replace('.', '_');
```

Comment:
This doesn't seem like a good default for a server in particular since multiple instances will all have different metric names and be hard to graph on a dashboard.